src/views/admin-portal/admin-portal.vue

```diff
@@ -0,0 +1,236 @@
+import Header from '@/components/header/header-component.vue';
+import { DEFAULT_VENUE } from '@/services/venue-service';
+import { Venue } from '@/models/venue-model';
+import ApiService from '@/services/api-service';
+import { Watch } from 'vue-property-decorator';
+
+@Component({
+  components: {
+    Button,
+    Header,
+    AlertBanner,
+  },
+  props: {
+    allVenues: Array,
+    venueId: String,
+  },
+})
+export default class AdminPortal extends Vue {
+  public apiService = new ApiService();
+  public venue: Venue = DEFAULT_VENUE;
+  public venueId!: string;
+  public allVenues!: Venue[];
+  public bannerActive: boolean = false;
+  public bannerMessage: string = 'Changes Saved!';
+  public bannerColor: string = 'green';
+  public imageChanged: boolean = false;
+  public newImage: any;
+  public newImageToUpload: any;
+
+  @Watch('allVenues')
+  public handleUpdate() {
+    const copy = this.allVenues.find((venue) => venue.id === this.venueId);
+    const freshCopy = Object.assign({}, copy);
+    this.venue = freshCopy;
+  }
+
+  @Watch('bannerActive')
+  public promptSuccessMessage(value: boolean) {
+    if (value) {
+      window.scrollTo(0, 0);
+      setTimeout(() => {
+        this.bannerActive = false;
+        this.bannerColor = 'green';
+        this.bannerMessage = 'Changes Saved!';
+      }, 3500);
+    }
+  }
+
+  public async beforeMount() {
+    // Needed for hard refresh of page.
+    const copy = this.allVenues.find((venue) => venue.id === this.venueId);
+    const freshCopy = Object.assign({}, copy);
+    this.venue = freshCopy;
+  }
+
+  get validAddress() {
+    return !!this.venue.address;
+  }
+
+  get validName() {
+    return !!this.venue.name;
+  }
+
+  get validForm() {
+    return this.validAddress && this.validName;
+  }
+
+  get validImage() {
+    // Checks to see if the new image is a blob
+    return this.newImage.includes('blob');
+  }
+
+  get displayImage() {
+    return this.imageChanged
+      ? this.newImage
+      : this.venue.company_img_url
+      ? this.venue.company_img_url
+      : require('../../images/company-images/emptyVenue.jpeg');
+  }
+
+  public imageUpdated(e: Event) {
+    const target = e.target as HTMLInputElement;
+    this.newImageToUpload = (target.files as FileList)[0];
+    this.newImage = URL.createObjectURL((target.files as FileList)[0]);
+    this.imageChanged = true;
+  }
+
+  public async updateVenue() {
+    if (!this.validForm) {
+      this.bannerMessage = 'Please complete required fields.';
+      this.bannerColor = 'red';
+      this.bannerActive = true;
+    } else {
+      const response = await this.apiService.updateVenue(this.venue);
+      if (response) {
+        this.bannerActive = true;
+        this.venue = response.data.attributes;
+      }
+    }
+  }
+
+  public async submitImage() {
+    if (!this.validImage) {
+      this.bannerMessage = 'Please provide valid image.';
+      this.bannerColor = 'red';
```

Comment:
if i just click "submit image" as the first thing i do on this page, it doesn't do anything, so i think one of the checks here isn't happening